crates/oxc_semantic/src/builder.rs

```diff
@@ -398,6 +403,8 @@ impl<'a> Visit<'a> for SemanticBuilder<'a> {
 
 impl<'a> SemanticBuilder<'a> {
     fn enter_kind(&mut self, kind: AstKind<'a>) {
+        // let parent_node = self.nodes.parent_node(self.current_node_id);
+        self.class.enter_kind(kind, self.current_node_id, &self.nodes);
```

Comment:
I think it's better to move and slit these into code blow so we get 1 less AST node check.

Remember, the code is called on everything single ast node - every cpu instruction counts.